UnityProjects/MRTKDevTemplate/Assets/Scripts/KeyboardAudio.cs

```diff
@@ -0,0 +1,41 @@
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace Microsoft.MixedReality.Toolkit.UX
+{
+    /// <summary>
+    /// Adds touch events to the NonNativeKeyboard buttons (and a tap sound)
+    /// </summary>
+    public class KeyboardAudio : MonoBehaviour
+    {
+        [SerializeField]
+        private AudioClip clickSound = null;
+
+        private AudioSource clickSoundPlayer;
+
+        private void Start()
+        {
+            EnableTouch();
+        }
+
+        private void EnableTouch()
+        {
+            clickSoundPlayer = gameObject.AddComponent<AudioSource>();
```

Comment:
Let make AudioSource as a requirement, and call GetComponent<> instead.

Adding a component at runtime can be costly (if done a lot ... so I just learned)